lib/src/entrypoint.dart

```diff
@@ -167,10 +167,14 @@ class Entrypoint {
   ///
   /// Throws a [DataError] if the `.dart_tool/package_config.json` file isn't
   /// up-to-date relative to the pubspec and the lockfile.
-  PackageGraph get packageGraph => _packageGraph ??= _createPackageGraph();
-
-  PackageGraph _createPackageGraph() {
-    ensureUpToDate(); // TODO, really?
+  Future<PackageGraph> get packageGraph =>
+      _packageGraph ??= _createPackageGraph();
```

Comment:
I think it might be possible to do, but it could also be too smart :rofl: 

```suggestion
  late final packageGraph = () async {
    await ensureUpToDate();
    var packages = {
      for (var packageEntry in packageConfig.nonInjectedPackages)
        packageEntry.name: Package.load(
          packageEntry.name,
          packageEntry.resolvedRootDir(packageConfigPath),
          cache.sources,
        ),
    };
    packages[root.name] = root;
    return PackageGraph(this, packages);
  }();
```

Reply:
Unfortunately we initialize _packageGraph directly in some of our constructors. So I'm not sure we can make this work